libsolidity/ast/Types.cpp

```diff
@@ -2564,10 +2564,14 @@ FunctionType::FunctionType(VariableDeclaration const& _varDecl):
 				if (auto arrayType = dynamic_cast<ArrayType const*>(member.type))
 					if (!arrayType->isByteArray())
 						continue;
-				m_returnParameterTypes.push_back(TypeProvider::withLocationIfReference(
-					DataLocation::Memory,
-					member.type
-				));
+				if (auto referenceType = dynamic_cast<ReferenceType const*>(member.type))
+					m_returnParameterTypes.push_back(TypeProvider::withLocation(
+						referenceType,
+						DataLocation::Memory,
+						true
```

Comment:
Please someone confirm, that this should in fact always be pointers under any circumstances :-).

Reply:
The distinction between pointer and non-pointer is only relevant for storage types.

Reply:
Or phrased differently: Memory references should always be pointers.

Reply:
Ok, if that's so, then we should probably rather change ``copyForLocation``, ``withLocation``, ``withLocationIfReference``, etc., to always make the types pointers, if in memory...

Reply:
But yeah, thinking about it, I don't know when it should be different, so I'll look into changing those parts instead.